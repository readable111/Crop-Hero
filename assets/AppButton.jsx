import { useCallBack } from 'react';
import { 
	StyleSheet, 
	Text, 
	Image, 
	TouchableOpacity
} from 'react-native'

{/*creates a custom button that changes opacity when touched, allows user to specify text style & background color*/}
const AppButton = ({ onPress, title='Press me', icon, specifiedStyle, backgroundColor='' }) => {
	{/*containerless button with/without title and without icon*/}
	if (!backgroundColor && !icon) {
		return  (
			<TouchableOpacity activeOpacity={0.5} onPress={onPress}>
				  <Text style={specifiedStyle}>{title}</Text>
			</TouchableOpacity>
		)
	} 
	else if (backgroundColor && title && !icon) { {/*container button with title and without icon*/}
		return  (
			<TouchableOpacity activeOpacity={0.5} onPress={onPress} style={[styles.appButtonContainer, backgroundColor && {backgroundColor}]}>
				<Text style={specifiedStyle}>{title}</Text>
			</TouchableOpacity>
		)
	}
	else if (icon && !title) { {/*button without title and with icon*/}
		return  (
			<TouchableOpacity activeOpacity={0.5} onPress={onPress}>
					<Image 
						style={specifiedStyle}
						source={icon}
					/>
			</TouchableOpacity>
		)
	}
	else { {/*default to containerless, no icon*/}
		return  (
			<TouchableOpacity activeOpacity={0.5} onPress={onPress}>
				  <Text style={specifiedStyle}>{title}</Text>
			</TouchableOpacity>
		)
	}
}

const styles = StyleSheet.create({
	appButtonContainer: {
		elevation: 8,
		borderColor: "black",
		borderWidth: 2,
		borderRadius: 50,
		paddingHorizontal: 12,
		marginTop: 12,
		width: 172,
		height: 35,
		alignItems: 'center',
		justifyContent: 'center',
	}
})

export default AppButton;