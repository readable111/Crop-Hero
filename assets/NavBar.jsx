import { useState } from 'react';
import { 
	StyleSheet, 
	View, 
	Text, 
	StatusBar, 
	Image, 
	Alert,
    Dimensions
} from 'react-native'
import { useFonts } from 'expo-font'
import { router } from 'expo-router'
import { Col, Row } from './Grid.jsx'
import Colors from './Color.js'
import AppButton from './AppButton.jsx'

const NavBar = ({ homeSelected=false, cropsSelected=false, notebookSelected=false, dataHubSelected=false, profileSelected=false}) => {

    const [fontsLoaded, fontError] = useFonts({
        'WorkSans-Semibold': require('./fonts/WorkSans-SemiBold.ttf'),
        'Domine-Medium': require('./fonts/Domine-Medium.ttf'),
        'Domine-Regular': require('./fonts/Domine-Regular.ttf'),
    });

    {/*return an error if the fonts fail to load*/}
    if (!fontsLoaded && !fontError) {
        return null;
    }

	return  (
        <View style={{height: 100, width: '100%', alignItems: 'center', marginTop: 'auto'}}>
            <View style={styles.navbarCircle}></View>
            <View style={styles.btnGridContainer}>
                <Row height={35}>
                        <Col relativeColsCovered={4} alignItems='center'>
                            <AppButton title="" mci="grass" mciSize={35} mciColor={cropsSelected ? Colors.MALACHITE : Colors.SOFT_GREEN} onPress={() => router.replace('/crops')}/>
                        </Col>
                        <Col relativeColsCovered={4} alignItems='center'>
                            <AppButton title="" mci={notebookSelected ? "notebook-edit" : "notebook-edit-outline"} mciSize={30} mciColor={notebookSelected ? Colors.MALACHITE : Colors.SOFT_GREEN} onPress={() => router.replace('/notebook')}/>
                        </Col>
                        <Col relativeColsCovered={4} alignItems='center'>
                            <AppButton title="" mci={homeSelected ? "home" : "home-outline"} mciSize={35} mciColor={homeSelected ? Colors.MALACHITE : Colors.SOFT_GREEN} onPress={() => router.replace('/home')}/>
                        </Col>
                        <Col relativeColsCovered={4} alignItems='center'>
                            <AppButton title="" mci={dataHubSelected ? "chart-box" : "chart-box-outline"} mciSize={35} mciColor={dataHubSelected ? Colors.MALACHITE : Colors.SOFT_GREEN} onPress={() => Alert.alert('Test')}/>
                        </Col>
                        <Col relativeColsCovered={4} alignItems='center'>
                            <AppButton title="" mci={profileSelected ? "account" : "account-outline"} mciSize={35} mciColor={profileSelected ? Colors.MALACHITE : Colors.SOFT_GREEN} onPress={() => router.replace('/profile')}/>
                        </Col>
                </Row>
                <Row height={25}>
                    <Col relativeColsCovered={4} alignItems='center'>
                        <AppButton title="Crops" specifiedStyle={cropsSelected ? styles.selectedText : styles.unselectedText} onPress={() => router.replace('/crops')}/>
                    </Col>
                    <Col relativeColsCovered={4} alignItems='center'>
                        <AppButton title="Notebook" specifiedStyle={notebookSelected ? styles.selectedText : styles.unselectedText} onPress={() => router.replace('/notebook')}/>
                    </Col>
                    <Col relativeColsCovered={4} alignItems='center'>
                        <AppButton title="Home" specifiedStyle={homeSelected ? styles.selectedText : styles.unselectedText} onPress={() => router.replace('/home')}/>
                    </Col>
                    <Col relativeColsCovered={4} alignItems='center'>
                        <AppButton title="Data Hub" specifiedStyle={dataHubSelected ? styles.selectedText : styles.unselectedText} onPress={() => Alert.alert('Test')}/>
                    </Col>
                    <Col relativeColsCovered={4} alignItems='center'>
                        <AppButton title="Profile" specifiedStyle={profileSelected ? styles.selectedText : styles.unselectedText} onPress={() => router.replace('/profile')}/>
                    </Col>
                </Row>
            </View>
        </View>
	)
}

{/*define all of the custom styles for this page*/}
const styles = StyleSheet.create({
	//style for the grid layout
	btnGridContainer: {
		flex: 20, // # of columns
    	marginHorizontal: "auto",
    	width: '100%',
        backgroundColor: Colors.ALMOND_TAN,
        height: 50,
	},
    selectedText: {
        color: Colors.MALACHITE,
        fontSize: 14,
        fontFamily: 'Domine-Regular',
    },
    unselectedText: {
        color: Colors.SOFT_GREEN,
        fontSize: 14,
        fontFamily: 'Domine-Regular',
    },
    navbarCircle: {
		width: 96,
    	height: 96,
    	borderRadius: 96 / 2, //borderRadius cannot exceed 50% of width or React-Native makes it weird
		backgroundColor: Colors.ALMOND_TAN,
		alignItems: 'center',
        justifyContent: 'center',
        marginBottom: -64,
	},
})

export default NavBar;