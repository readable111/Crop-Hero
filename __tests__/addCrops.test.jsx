/****
 * @author Isaac Boodt, Daniel Moreno
 * @reviewer 
 * @tester 
 ***/

import { render, fireEvent, screen} from '@testing-library/react-native'
import { Input } from 'react-native'
import AddCrops from '../app/addcrops'
import React from 'react';
import AppButton from '../assets/AppButton.jsx';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { router } from 'expo-router';
import NavBar from '../assets/NavBar.jsx';
import Colors from '../assets/Color';
import * as Font from 'expo-font';


jest.mock('expo-router', () => ({
    router:{
        push: jest.fn(),
    },
}
))

jest.mock('expo-font', () => ({
    useFonts: jest.fn(),
    loadAsync: jest.fn().mockResolvedValue(true),
    isLoaded: jest.fn(()=>true)
}))


describe('<AddCrops/>', () =>{
    beforeEach(async () => {
        await AsyncStorage.setItem("dark_mode_setting", false);

        Font.useFonts.mockReturnValue([true, false])
        
        const [fontsLoaded, fontError] = Font.useFonts()
        expect(fontsLoaded).toBe(true)
        expect(fontError).toBe(false)

    })
    /*
    it('render input with placeholder, handles text change', () => {
        const handleChange = jest.fn();

        const { getByTestId} = 
        render(<addCrops handleChange={handleChange}/>);

        const nameInput = getByTestId("name-input")

        fireEvent.changeText(nameInput, Apple);

        expect(handleChange).toHaveBeenCalledWith('name', 'Apple');
    })
        */
    test('renders correctly', () =>{
        const tree = render(<AddCrops/>).toJSON();
        expect(tree).toMatchSnapshot();
    })
})
/*
describe('<addCrops/>', () =>{
    it('render input with placeholder, handles text change', () => {
        const handleChange = jest.fn();

        const { getByTestId} = 
        render(<addCrops handleChange={handleChange}/>);

        const nameInput = getByTestId("name-input")

        fireEvent.changeText(nameInput, Apple);

        expect(handleChange).toHaveBeenCalledWith('name', 'Apple');
    })
})
    */