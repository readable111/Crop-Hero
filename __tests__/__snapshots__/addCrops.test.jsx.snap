// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCrops/> renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      [
        {
          "backgroundColor": "#97A5BF",
          "height": "100%",
        },
        false,
      ]
    }
  >
    <View
      style={
        [
          {
            "backgroundColor": "#F1DDBF",
            "borderColor": "#1E1E1E",
            "borderWidth": 1,
            "padding": 18,
          },
          false,
        ]
      }
    >
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "position": "relative",
          }
        }
      >
        <Text
          style={
            [
              {
                "alignContent": "center",
                "flex": 1,
                "fontFamily": "Domine-Medium",
                "fontSize": 42,
                "textAlign": "right",
              },
              false,
            ]
          }
        >
          Add Crop
        </Text>
        <View
          style={
            {
              "backgroundColor": "#00DC40",
              "borderBottomLeftRadius": 140,
              "borderBottomRightRadius": 140,
              "height": 70,
              "left": 0,
              "overflow": "hidden",
              "position": "absolute",
              "top": -19,
              "width": 140,
            }
          }
        />
      </View>
    </View>
    <View>
      <View
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#00DC40",
            "borderRadius": 20,
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 370,
            "marginTop": 10,
            "position": "absolute",
            "width": 40,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID=""
        >
          <Text
            allowFontScaling={false}
            selectable={false}
            style={
              [
                {
                  "color": "#1E1E1E",
                  "fontSize": 30,
                },
                undefined,
                {
                  "fontFamily": "material-community",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                {},
              ]
            }
          >
            ó°†“
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "height": 40,
            "justifyContent": "center",
            "marginLeft": 10,
            "width": 40,
          }
        }
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "opacity": 1,
            }
          }
          testID=""
        >
          <Image
            source={1}
            style={{}}
          />
        </View>
      </View>
    </View>
    <RCTScrollView>
      <View>
        <View
          style={
            {
              "marginTop": 20,
            }
          }
        />
        <Text
          style={
            [
              {
                "alignSelf": "flex-start",
                "backgroundColor": "white",
                "borderColor": "white",
                "fontFamily": "Domine-Regular",
                "fontSize": 16,
                "marginLeft": "15%",
                "marginTop": -15,
                "zIndex": 10,
              },
              false,
            ]
          }
        >
          Crop Name
        </Text>
        <View
          style={
            {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 2,
                "borderColor": "#1E1E1E",
                "borderRadius": 12,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 40,
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": -5,
                "overflow": "hidden",
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": "90%",
                "zIndex": 1,
              }
            }
          >
            <TextInput
              editable={true}
              maxLength={128}
              onChangeText={[Function]}
              placeholder="name"
              placeholderTextColor="#86939e"
              style={
                {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="name-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <Text
          style={
            [
              {
                "alignSelf": "flex-start",
                "backgroundColor": "white",
                "borderColor": "white",
                "fontFamily": "Domine-Regular",
                "fontSize": 16,
                "marginLeft": "15%",
                "marginTop": -15,
                "zIndex": 10,
              },
              false,
            ]
          }
        >
          Variety
        </Text>
        <View
          style={
            {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 2,
                "borderColor": "#1E1E1E",
                "borderRadius": 12,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 40,
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": -5,
                "overflow": "hidden",
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": "90%",
                "zIndex": 1,
              }
            }
          >
            <TextInput
              editable={true}
              maxLength={128}
              onChangeText={[Function]}
              placeholder=" Variety"
              placeholderTextColor="#86939e"
              style={
                {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <Text
          style={
            [
              {
                "alignSelf": "flex-start",
                "backgroundColor": "white",
                "borderColor": "white",
                "fontFamily": "Domine-Regular",
                "fontSize": 16,
                "marginLeft": "15%",
                "marginTop": -15,
                "zIndex": 10,
              },
              false,
            ]
          }
        >
          Source
        </Text>
        <View
          style={
            {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 2,
                "borderColor": "#1E1E1E",
                "borderRadius": 12,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 40,
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": -5,
                "overflow": "hidden",
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": "90%",
                "zIndex": 1,
              }
            }
          >
            <TextInput
              editable={true}
              maxLength={128}
              onChangeText={[Function]}
              placeholder=" Source"
              placeholderTextColor="#86939e"
              style={
                {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <Text
          style={
            [
              {
                "alignSelf": "flex-start",
                "backgroundColor": "white",
                "borderColor": "white",
                "fontFamily": "Domine-Regular",
                "fontSize": 16,
                "marginLeft": "15%",
                "marginTop": -15,
                "zIndex": 10,
              },
              false,
            ]
          }
        >
          Date Planted
        </Text>
        <View
          style={
            {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 2,
                "borderColor": "#1E1E1E",
                "borderRadius": 12,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 40,
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": -5,
                "overflow": "hidden",
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": "90%",
                "zIndex": 1,
              }
            }
          >
            <TextInput
              editable={true}
              maxLength={10}
              onChangeText={[Function]}
              placeholder=" Date Planted"
              placeholderTextColor="#86939e"
              style={
                {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <Text
          style={
            [
              {
                "alignSelf": "flex-start",
                "backgroundColor": "white",
                "borderColor": "white",
                "fontFamily": "Domine-Regular",
                "fontSize": 16,
                "marginLeft": "15%",
                "marginTop": -15,
                "zIndex": 10,
              },
              false,
            ]
          }
        >
          Location
        </Text>
        <View
          style={
            {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 2,
                "borderColor": "#1E1E1E",
                "borderRadius": 12,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 40,
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": -5,
                "overflow": "hidden",
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": "90%",
                "zIndex": 1,
              }
            }
          >
            <TextInput
              editable={true}
              maxLength={128}
              onChangeText={[Function]}
              placeholder=" Location"
              placeholderTextColor="#86939e"
              style={
                {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <Text
          style={
            [
              {
                "alignSelf": "flex-start",
                "backgroundColor": "white",
                "borderColor": "white",
                "fontFamily": "Domine-Regular",
                "fontSize": 16,
                "marginLeft": "15%",
                "marginTop": -15,
                "zIndex": 10,
              },
              false,
            ]
          }
        >
          Comments
        </Text>
        <View
          style={
            {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 2,
                "borderColor": "#1E1E1E",
                "borderRadius": 12,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 40,
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": -5,
                "overflow": "hidden",
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": "90%",
                "zIndex": 1,
              }
            }
          >
            <TextInput
              editable={true}
              maxLength={1024}
              onChangeText={[Function]}
              placeholder=" Comments"
              placeholderTextColor="#86939e"
              style={
                {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <Text
          style={
            [
              {
                "alignSelf": "flex-start",
                "backgroundColor": "white",
                "borderColor": "white",
                "fontFamily": "Domine-Regular",
                "fontSize": 16,
                "marginLeft": "15%",
                "marginTop": -15,
                "zIndex": 10,
              },
              false,
            ]
          }
        >
          Started Indoors?
        </Text>
        <View
          style={
            {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 2,
                "borderColor": "#1E1E1E",
                "borderRadius": 12,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 40,
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": -5,
                "overflow": "hidden",
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": "90%",
                "zIndex": 1,
              }
            }
          >
            <TextInput
              editable={true}
              maxLength={3}
              onChangeText={[Function]}
              placeholder=" Indoors? (Y/N)"
              placeholderTextColor="#86939e"
              style={
                {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <Text
          style={
            [
              {
                "alignSelf": "flex-start",
                "backgroundColor": "white",
                "borderColor": "white",
                "fontFamily": "Domine-Regular",
                "fontSize": 16,
                "marginLeft": "15%",
                "marginTop": -15,
                "zIndex": 10,
              },
              false,
            ]
          }
        >
          Active
        </Text>
        <View
          style={
            {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 2,
                "borderColor": "#1E1E1E",
                "borderRadius": 12,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 40,
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": -5,
                "overflow": "hidden",
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": "90%",
                "zIndex": 1,
              }
            }
          >
            <TextInput
              editable={true}
              maxLength={3}
              onChangeText={[Function]}
              placeholder=" Active"
              placeholderTextColor="#86939e"
              style={
                {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <Text
          style={
            [
              {
                "alignSelf": "flex-start",
                "backgroundColor": "white",
                "borderColor": "white",
                "fontFamily": "Domine-Regular",
                "fontSize": 16,
                "marginLeft": "15%",
                "marginTop": -15,
                "zIndex": 10,
              },
              false,
            ]
          }
        >
          Type
        </Text>
        <View
          style={
            {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 2,
                "borderColor": "#1E1E1E",
                "borderRadius": 12,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 40,
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": -5,
                "overflow": "hidden",
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": "90%",
                "zIndex": 1,
              }
            }
          >
            <TextInput
              editable={true}
              maxLength={64}
              onChangeText={[Function]}
              placeholder=" Type"
              placeholderTextColor="#86939e"
              style={
                {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <Text
          style={
            [
              {
                "alignSelf": "flex-start",
                "backgroundColor": "white",
                "borderColor": "white",
                "fontFamily": "Domine-Regular",
                "fontSize": 16,
                "marginLeft": "15%",
                "marginTop": -15,
                "zIndex": 10,
              },
              false,
            ]
          }
        >
          Medium
        </Text>
        <View
          style={
            {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 2,
                "borderColor": "#1E1E1E",
                "borderRadius": 12,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 40,
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": -5,
                "overflow": "hidden",
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": "90%",
                "zIndex": 1,
              }
            }
          >
            <TextInput
              editable={true}
              maxLength={64}
              onChangeText={[Function]}
              placeholder=" Medium"
              placeholderTextColor="#86939e"
              style={
                {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <Text
          style={
            [
              {
                "alignSelf": "flex-start",
                "backgroundColor": "white",
                "borderColor": "white",
                "fontFamily": "Domine-Regular",
                "fontSize": 16,
                "marginLeft": "15%",
                "marginTop": -15,
                "zIndex": 10,
              },
              false,
            ]
          }
        >
          HRF Number
        </Text>
        <View
          style={
            {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 2,
                "borderColor": "#1E1E1E",
                "borderRadius": 12,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 40,
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": -5,
                "overflow": "hidden",
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": "90%",
                "zIndex": 1,
              }
            }
          >
            <TextInput
              editable={true}
              maxLength={64}
              onChangeText={[Function]}
              placeholder=" HRF Number"
              placeholderTextColor="#86939e"
              style={
                {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <Text
          style={
            [
              {
                "alignSelf": "flex-start",
                "backgroundColor": "white",
                "borderColor": "white",
                "fontFamily": "Domine-Regular",
                "fontSize": 16,
                "marginLeft": "15%",
                "marginTop": -15,
                "zIndex": 10,
              },
              false,
            ]
          }
        >
          Yield
        </Text>
        <View
          style={
            {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 2,
                "borderColor": "#1E1E1E",
                "borderRadius": 12,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 40,
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": -5,
                "overflow": "hidden",
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": "90%",
                "zIndex": 1,
              }
            }
          >
            <TextInput
              editable={true}
              maxLength={64}
              onChangeText={[Function]}
              placeholder=" Yield"
              placeholderTextColor="#86939e"
              style={
                {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
        <Text
          style={
            [
              {
                "alignSelf": "flex-start",
                "backgroundColor": "white",
                "borderColor": "white",
                "fontFamily": "Domine-Regular",
                "fontSize": 16,
                "marginLeft": "15%",
                "marginTop": -15,
                "zIndex": 10,
              },
              false,
            ]
          }
        >
          Visible
        </Text>
        <View
          style={
            {
              "paddingHorizontal": 10,
              "width": "100%",
            }
          }
        >
          <View
            collapsable={false}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "white",
                "borderBottomWidth": 2,
                "borderColor": "#1E1E1E",
                "borderRadius": 12,
                "borderWidth": 2,
                "flexDirection": "row",
                "height": 40,
                "marginLeft": "5%",
                "marginRight": "5%",
                "marginTop": -5,
                "overflow": "hidden",
                "transform": [
                  {
                    "translateX": 0,
                  },
                ],
                "width": "90%",
                "zIndex": 1,
              }
            }
          >
            <TextInput
              editable={true}
              maxLength={64}
              onChangeText={[Function]}
              placeholder=" Visibility"
              placeholderTextColor="#86939e"
              style={
                {
                  "color": "#242424",
                  "flex": 1,
                  "fontSize": 16,
                  "minHeight": 40,
                }
              }
              testID="RNE__Input__text-input"
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            style={
              {
                "color": "#ff190c",
                "fontSize": 12,
                "margin": 5,
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
    <View />
  </View>
</View>
`;
